crates/accelerate/src/sabre_swap/mod.rs

```diff
@@ -262,20 +330,42 @@ fn swap_map_trial(
     let mut num_search_steps: u8 = 0;
     let mut qubits_decay: Vec<f64> = vec![1.; num_qubits];
     let mut rng = Pcg64Mcg::seed_from_u64(seed);
+    let mut node_block_results: HashMap<usize, Vec<BlockResult>> =
+        HashMap::with_capacity(dag.node_blocks.len());
 
     for node in dag.dag.node_indices() {
         for edge in dag.dag.edges(node) {
             required_predecessors[edge.target().index()] += 1;
         }
     }
+
+    // This lambda is used to curry parameters so we can avoid
```

Comment:
Super Nit (in this context they're really interchangable though :D):

```suggestion
    // This closure is used to curry parameters so we can avoid
```